### DISCUSSION TITLE
Signature Help in Intelephense (php) crashes

### DISCUSSION THREAD

**Discussion by yngwi** [folke/noice.nvim#263] [Q&A]
Hi, in my setup, nvim crashes whenever I open the signature help for the Intelephense PHP language server, either with the automated variant using `lsp.signature = true` or triggering `vim.lsp.buf.signature_help()`. As far as I can tell, other language servers work ok and the other LSP providers (diagnostics, documentation and so on) also seem to work just fine in Intelephense. If I disable noice, the signature help also seems to work just fine. What could be the reason for this? Thanks!

**Comment by folke** [comment]
No idea, sorry. You'll have to investigate yourself. Noice doesn't do anything that can crash Neovim, so must be a bug somewhere in core triggering it. Really no idea. Also not using intelephense

**Reply by yngwi** [reply]
Hi, thanks for the reply. I have tried to narrow it down a little and seems as if the problem appears in combination of treesitter/noice/Intelephense. See this small config below, it works with either treesitter or noice present, not both. In the state as written below, with both active, as soon as the popup for the signature help appears, nvim freezes while the popup content stays empty. I have added two screenshots below, the first without noice, the second with it, frozen. Could it be caused by the content of the Intelephense signature help text?

![image](https://user-images.githubusercontent.com/963440/207268316-5676dd36-4cb9-4ccc-b9ac-5ce42f3c2b3f.png)

![image](https://user-images.githubusercontent.com/963440/207267723-620464c8-f1ed-4d39-b788-44a7bddac6de.png)


```vim
call plug#begin('~/.vim/plugged')
Plug 'neovim/nvim-lspconfig'
Plug 'MunifTanjim/nui.nvim'
Plug 'rcarriga/nvim-notify'
Plug 'folke/noice.nvim'
Plug 'nvim-treesitter/nvim-treesitter', {'do': ':TSUpdate'}
call plug#end()

lua << EOF
vim.opt.clipboard = "unnamedplus"
vim.opt.termguicolors = true

require("lspconfig").intelephense.setup({
	init_options = {
		licenceKey = vim.fn.expand("$HOME/.intelephense/license.txt"),
		storagePath = vim.fn.expand("$HOME/.intelephense"),
		globalStoragePath = vim.fn.expand("$HOME/.intelephense"),
	},
	settings = {
		intelephense = {
			environment = {
				phpVersion = "8.1",
			},
		},
	},
})

require("noice").setup({})

vim.api.nvim_set_keymap('n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', { noremap = true, silent = true })
EOF

```
